### ISSUE TITLE
[Musicxml Export] - Confusing staff-distance

### ISSUE THREAD

**Issue by hhpmusic** [musescore/MuseScore#22098]
### Issue type

Import/export issue

### Bug description

[hidden staves.zip](https://github.com/musescore/MuseScore/files/14800219/hidden.staves.zip)

### Steps to reproduce

In this example, I'm demonstrating various hidden staves for Musicxml export. Since Musescore still doesn't use part-object print-object="yes/no", I have to evaluate the current staff-distance to see if it fulfills the condition of implementing correct braille exsport. The result is somewhat tricky, so I'm wondering if we still need to stick on this confusing tag.
From the exported Musicxml, I learned a positive value means the staff (except the first staff, which is controlled by staff 2's value) is displayed, while a negative value means it's hidden. The problem occurs on the zero value. At the first glance, I thought it means the upper staff is hidden, and the current staff can have notes. Please see oboe part at bar 5 and 9, and violin staff 1 bar 1, where the upper staff does hide. However, at bar 1, violin 2 staff has the value 83, where the first violin staff is actually hidden. So, how can we know which staff is hidden in this case? Does this value confusing? If yes, we should abandon it and use the staff-details print-object="yes/no", otherwise the output, both musicxml import and braille conversion, will bring big confusion. Am I right? Or is there anyone to point out the correct interpretation of this value? This, and the instrument changes, are two blocking stones for my braille transcription, so I hope to make things clear and ask our software developers to implement correct braille output, before we end the dAISY Music Braille Project next year. Thanks!

Haipeng

### MuseScore Version

4.2

### Regression

No.

### Operating system

Windows

**Comment by zacjansheski** [comment]
@shoogle please take a look

**Comment by lvinken** [comment]
The current MusicXML exporter does not handle hidden staves at all. This results in incorrect staff-distance calculations:

- A zero staff-distance is generated when the preceding staff (in the same system) is hidden
- A staff-distance of -40 is generated when the preceding staff is not hidden but the staff itself is hidden

 The issue is located in the function ExportMusicXml::print().

**Comment by hhpmusic** [comment]
> The current MusicXML exporter does not handle hidden staves at all. This results in incorrect staff-distance calculations:
> 
>     * A zero staff-distance is generated when the preceding staff (in the same system) is hidden
> 
>     * A staff-distance of -40 is generated when the preceding staff is not hidden but the staff itself is hidden
> 
> 
> The issue is located in the function ExportMusicXml::print().
I agree, this export is completely wrong. Sometimes a hidden staff is 0, sometimes not, very strange. The export is quite inaccurate. When I tried to tell the developers to figure out which kind of number stands for which case, I was totally confused by the variable result. Musescore is like a naughty magician!

Haipeng

**Comment by lvinken** [comment]
File [hidden.staves.MS444.current.and.corrected.zip](https://github.com/user-attachments/files/18270848/hidden.staves.MS444.current.and.corrected.zip) contains both the current MuseScore 4.4.4 xml export and a manually corrected version, that imports correctly into Finale NotePad. Note that some distances may be somewhat incorrect, as these were not calculated by MuseScore by guessed by me. Would this version be OK for you ?

Regards, Leon.

**Comment by hhpmusic** [comment]
> File [hidden.staves.MS444.current.and.corrected.zip](https://github.com/user-attachments/files/18270848/hidden.staves.MS444.current.and.corrected.zip) contains both the current MuseScore 4.4.4 xml export and a manually corrected version, that imports correctly into Finale NotePad. Note that some distances may be somewhat incorrect, as these were not calculated by MuseScore by guessed by me. Would this version be OK for you ?
> 
> Regards, Leon.
This is better, thanks! But the most accurate way is still staff-details print-object. My thought is to implement this instead of relying only on distances. All other notation software export hidden staves this way.

Haipeng

**Comment by lvinken** [comment]
As far as I know the corrected file includes correct "staff-details print-object" elements. In case you find any mistakes, please let me know.

**Comment by hhpmusic** [comment]
Sorry, I missed that. This example is correct.

**Comment by lvinken** [comment]
Created a few simple test files, see https://github.com/lvinken/MuseScore/tree/issue_22098_testdata
